airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -310,6 +325,29 @@ else if (hasFailed.get()) {
 
   }
 
+  private class ThreadedTimeHolder {
+
+    private long sourceReadEndTime = -1;
+    private long destinationWriteEndTime = -1;
+
+    public synchronized void trackSourceReadEndTime() {
+      this.sourceReadEndTime = System.currentTimeMillis();
+    }
+
+    public synchronized void trackDestinationWriteEndTime() {
+      this.destinationWriteEndTime = System.currentTimeMillis();
+    }
+
+    public synchronized long getSourceReadEndTime() {
+      return this.sourceReadEndTime;
+    }
+
+    public synchronized long getDestinationWriteEndTime() {
```

Comment:
for my understanding, why does this need to be synchronised? Is it accessed by multiple threads?

Reply:
ah nvm I see why now - both runnables need it!